static/js/topic_page/map_tile.tsx

```diff
@@ -0,0 +1,260 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for rendering a map type tile.
+ */
+
+import axios from "axios";
+import _ from "lodash";
+import React, { useEffect, useState } from "react";
+
+import { drawChoropleth, getColorScale } from "../chart/draw_choropleth";
+import { GeoJsonData } from "../chart/types";
+import { formatNumber } from "../i18n/i18n";
+import { StatApiResponse } from "../shared/stat_types";
+import { NamedPlace } from "../shared/types";
+import { getCappedStatVarDate, urlToDomain } from "../shared/util";
+import {
+  DataPointMetadata,
+  getPlaceChartData,
+  getTitle,
+} from "../tools/map/util";
+import {
+  GetStatSetResponse,
+  PlacePointStat,
+  StatMetadata,
+} from "../tools/shared_util";
+
+const CHART_HEIGHT = 400;
+const SVG_CONTAINER_ELEMENT: React.RefObject<HTMLDivElement> = React.createRef();
+
+interface MapTilePropType {
+  tileId: string;
+  placeDcid: string;
+  enclosedPlaceType: string;
+  statVarDcid: string;
+  chartTitle: string;
+  isUsaPlace: boolean;
+  denomDcid?: string;
+  unit?: string;
+  scaling?: number;
+}
+
+interface RawData {
+  geoJsonData: GeoJsonData;
+  placeStatData: PlacePointStat;
+  metadataMap: Record<string, StatMetadata>;
+  populationData: StatApiResponse;
+}
+
+interface ChartData {
+  dataValues: { [dcid: string]: number };
+  metadata: { [dcid: string]: DataPointMetadata };
+  sources: Set<string>;
+  geoJsonData: GeoJsonData;
+  chartTitle: string;
+}
+
+export function MapTile(props: MapTilePropType): JSX.Element {
+  const [rawData, setRawData] = useState<RawData | undefined>(null);
+  const [chartData, setChartData] = useState<ChartData | undefined>(null);
+  const sourcesJsx = chartData ? getSourcesJsx(chartData.sources) : [];
+
+  useEffect(() => {
```

Comment:
I really like the structure here to use "fetchData", "processData" and "drawMap". All other tiles can follow same pattern, which makes the maintenance easier.